search/routes/api/classic.py

```diff
@@ -28,18 +28,16 @@ def query() -> Response:
     # requested = request.accept_mimetypes.best_match([JSON, ATOM_XML])
     # if requested == ATOM_XML:
     #     return serialize.as_atom(data), status, headers
-    response = serialize.as_json(data['results'], query=data['query'])
-    response.status_code = status_code
-    response.headers.extend(headers)
-    return response
+    response_data = serialize.as_atom(data['results'], query=data['query'])
+    return response_data, status_code, headers # type: ignore
```

Comment:
You mentioned needing to implement content negotiation. I would actually recommend not doing that, and just adding the appropriate ``Content-type`` header on the response (getting ``Content-Type: text/html; charset=utf-8`` right now).

```suggestion
    response_data = serialize.as_atom(data['results'], query=data['query'])
    headers.update({'Content-type': 'application/atom+xml; charset=utf-8'})
    response: Response = make_response(response_data, status_code, headers)
    return response
```